### PR TITLE
feat(live): add a new resource to generate a sign URL

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new resource to generate a sign URL.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support a new resource to generate a sign URL.
2. 2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/live" TESTARGS="-run TestAccUrlAuthentication_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/live -v -run TestAccUrlAuthentication_basic -timeout 360m -parallel 4
=== RUN   TestAccUrlAuthentication_basic
=== PAUSE TestAccUrlAuthentication_basic
=== CONT  TestAccUrlAuthentication_basic
--- PASS: TestAccUrlAuthentication_basic (433.68s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      433.771s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/resources/live_url_authentication.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+page_title: "HuaweiCloud: huaweicloud_live_url_authentication"
+description: |-
+  Manages an URL authentication resource within HuaweiCloud.
+---
+
+# huaweicloud_live_url_authentication
+
+Manages an URL authentication resource within HuaweiCloud.
+
+-> The current resource is a one-time resource, and destroying this resource will not change the current status.
```
Comment (jinyangyang222) [inline docs/resources/live_url_authentication.md:13]: Does repeatedly applying this resource have any impact?

File: docs/resources/live_url_authentication.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+-> Before creating the resource, you need to cofiguration the URL validation first (creating
+  the URL validation resource first).
+  [documentation](https://support.huaweicloud.com/iLive-live/live_01_0049.html)
```
Comment (deer-hang) [inline docs/resources/live_url_authentication.md]: Please use the English document link.

Comment (deer-hang) [reply docs/resources/live_url_authentication.md]: -> Before creating the resource, you need to cofiguration the URL validation first (creating
  the URL validation resource first). Refer to
  [URL Validation](https://support.huaweicloud.com/intl/en-us/iLive-live/live_01_0049.html) for more details.

File: docs/resources/live_url_authentication.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  Changing this parameter will create a new resource.
+
+* `stream_name` - (Required, String, ForceNew) Specifies the stream name.
+  Changing this parameter will create a new resource.
+
+* `app_name` - (Required, String, ForceNew) Specifies the application name.
+  Changing this parameter will create a new resource.
+
+* `check_level` - (Optional, Int, ForceNew) Specifies the check level.
+  This parameter is valid and mandatory when the value of the `auth_type` in the URL validation is set to **c_aes**.
```
Comment (deer-hang) [inline docs/resources/live_url_authentication.md:61]:
```suggestion
  This parameter is valid and mandatory only when the signing method is **c_aes** in URL validation.
```

File: docs/resources/live_url_authentication.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  The valid values are as follows:
+  + **3**: Indicates only checks the LiveID matches or not but not the validity of the signed URL.
+  + **5**: Indicates checks the LiveID matches or not and the validity of timestamp.
```
Comment (deer-hang) [inline docs/resources/live_url_authentication.md]:
```suggestion
  This parameter is valid and mandatory only when the signing method is **c_aes** in URL validation.
  The valid values are as follows:
  + `3`: The system checks only LiveID but not the validity of the signed URL.
  + `5`: The system checks LiveID and the validity of timestamp.
```

File: docs/resources/live_url_authentication.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+  The value of the LiveID consists of `app_name` and `stream_name`: <app_name>/<stream_name>.
```
Comment (deer-hang) [inline docs/resources/live_url_authentication.md]:
```suggestion
  -> The value of the LiveID consists of `app_name` and `stream_name`: <app_name>/<stream_name>.
```

File: docs/resources/live_url_authentication.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  Changing this parameter will create a new resource.
+
+* `start_time` - (Optional, String, ForceNew) Specifies the start time of the valid access time defined by the user.
+  Changing this parameter will create a new resource.
+  The time is in UTC, the format is **yyyy-mm-ddThh:mm:ssZ**. e.g. **2024-06-01T15:03:01Z**
+  If this parameter is not specified or is left empty, the current time is used by default.
```
Comment (deer-hang) [inline docs/resources/live_url_authentication.md]:
```suggestion
  Defaults to current time.
```

File: docs/resources/live_url_authentication.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  The valid values are as follows:
+  + **3**: Indicates only checks the LiveID matches or not but not the validity of the signed URL.
+  + **5**: Indicates checks the LiveID matches or not and the validity of timestamp.
+
+  The value of the LiveID consists of `app_name` and `stream_name`: <app_name>/<stream_name>.
+  Changing this parameter will create a new resource.
+
+* `start_time` - (Optional, String, ForceNew) Specifies the start time of the valid access time defined by the user.
+  Changing this parameter will create a new resource.
+  The time is in UTC, the format is **yyyy-mm-ddThh:mm:ssZ**. e.g. **2024-06-01T15:03:01Z**
```
Comment (deer-hang) [inline docs/resources/live_url_authentication.md]:
```suggestion
  The time is in UTC, the format is **yyyy-mm-ddThh:mm:ssZ**, e.g. **2024-06-01T15:03:01Z**.
```

File: huaweicloud/services/acceptance/live/resource_huaweicloud_live_url_authentication_test.go [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  start_time  = "%[2]s"
+
+  depends_on = [huaweicloud_live_url_validation.test]
+}
+`, name, startTime)
+}
+
+func testAccUrlAuthentication_stream(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_live_domain" "try" {
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/live/resource_huaweicloud_live_url_authentication_test.go:76]: We usually use "test" as an alias.

File: huaweicloud/services/live/resource_huaweicloud_live_url_authentication.go [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	keyChain := utils.PathSearch("keychain", respBody, make([]interface{}, 0)).([]interface{})
+	if len(keyChain) == 0 {
+		return diag.Errorf("err creating URL authentication, the 'key_chain' not found in API response")
+	}
+
```
Comment (deer-hang) [inline huaweicloud/services/live/resource_huaweicloud_live_url_authentication.go:123]:
```
	resourceId, err := uuid.GenerateUUID()
	if err != nil {
		return diag.Errorf("unable to generate ID: %s", err)
	}

	d.SetId(resourceId)
```

After confirming that the resource has been created successfully, you can configure the resource ID.

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/live" TESTARGS="-run TestAccUrlAuthentication_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/live -v -run TestAccUrlAuthentication_basic -timeout 360m -parallel 4
=== RUN   TestAccUrlAuthentication_basic
=== PAUSE TestAccUrlAuthentication_basic
=== CONT  TestAccUrlAuthentication_basic
--- PASS: TestAccUrlAuthentication_basic (392.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      392.862s
```

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/live" TESTARGS="-run TestAccUrlAuthentication_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/live -v -run TestAccUrlAuthentication_basic -timeout 360m -parallel 4
=== RUN   TestAccUrlAuthentication_basic
=== PAUSE TestAccUrlAuthentication_basic
=== CONT  TestAccUrlAuthentication_basic
--- PASS: TestAccUrlAuthentication_basic (371.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      372.004s
```

**Pull request merged**